drivers/flash/Kconfig

```diff
@@ -23,6 +23,16 @@ config FLASH_HAS_PAGE_LAYOUT
 	  This option is enabled when the SoC flash driver supports
 	  retrieving the layout of flash memory pages.
 
+config FLASH_DEVICE_REQUIRES_ERASE
```

Comment:
"none of the devices require erase"

Reply:
I think this should be rather FLASH_ANY_FLASH_DEVICE_REQUIRES_ERASE or something like that, as a driver is expected to set that to indicate that such device, a device that requires erase prior to write, is compiled within system.